quinn-proto/src/connection.rs

```diff
@@ -1139,10 +1097,10 @@ impl Connection {
                                                 .into(),
                                         },
                                     ));
-                                    return State::handshake_failed(
+                                    return Ok(State::handshake_failed(
```

Comment:
@Ralith maybe I got this wrong during the TLS overhaul, should we be returning `TRANSPORT_PARAMETER_ERROR` here instead of `TLS_HANDSHAKE_FAILED`?

Reply:
I don't think so. I don't see this case called out specifically in the draft, but `TRANSPORT_PARAMETER_ERROR` describes malformed transport parameters, whereas their absence seems better conveyed by a TLS `missing_extension` alert.

Reply:
Which I now realize probably means that the event should be changed to be consistent.

Reply:
Cool, will update event to be `TransportError::TLS_HANDSHAKE_FAILED.into()` then.